components/input/Input.js

```diff
@@ -55,28 +55,24 @@ export default class Input extends Component {
     }
   }
 
-  onChange(value) {
-    if (this.isNumberInput) {
-      this.setState({ value: Number(value.target.value) });
-    } else {
-      this.setState({ value: value.target.value });
-    }
+  onChange(event) {
+    const value = event.target.value;
+
+    this.setState(state => ({
+      value: this.isNumberInput ? Number(value) : value,
+    }));
   }
 
   increaseValue() {
-    if (this.state.value) {
-      return this.setState({ value: this.state.value + this.props.step });
-    }
-
-    return this.setState({ value: this.props.step });
+    this.setState((state, props) => ({
+      value: state.value ? state.value + props.step : props.step,
```

Comment:
Cool, thanks. Can you use `prevState` as variable name though, to be more explicit? You get the previous state and the current props as arguments 👍

Reply:
✅ Done.